### PR TITLE
Do input checking on name.

### PR DESCRIPTION
**Pull request by arringtonh**

### CODE REVIEW

**Review by leeannachen — COMMENTED** [review]
lgtm

File: proj/src/main/webapp/chat.js [hunk 1]
```diff
@@ -754,7 +754,9 @@ async function addUsernameToMessage(uid, preview) {
     const userRef = firebase.database().ref('/users/'+uid);
     await userRef.once("value", snap => {
         if(snap.val()) {
-            preview.querySelector('#username').innerText = snap.val().firstName + ' ' + snap.val().lastName;
+            const fName = snap.val().firstName || ' ';
+            const lName = snap.val().lastName || ' ';
```
Comment (aokeke) [inline proj/src/main/webapp/chat.js:758]: Is there a reason you use a space instead of an empty string?

**Review by aokeke — APPROVED** [review]
LGTM

File: proj/src/main/webapp/chat.js (hunk 1 above)
Comment (arringtonh) [reply proj/src/main/webapp/chat.js:758]: no

**Pull request merged**